kube-runtime/src/utils.rs

```diff
@@ -137,3 +139,43 @@ where
     let via = make_via_stream(oks); // the map_ok/err function
     stream::select(via.into_stream(), errs.map(Err)) // recombine
 }
+
+/// A [`JoinHandle`] that cancels the [`Future`] when dropped, rather than detaching it
+pub struct CancelableJoinHandle<T> {
+    inner: JoinHandle<Result<T, Aborted>>,
+    abort_handle: AbortHandle,
+}
+
+impl<T> CancelableJoinHandle<T>
+where
+    T: Send + 'static,
+{
+    pub fn spawn(future: impl Future<Output = T> + Send + 'static, runtime: &Handle) -> Self {
+        let (abortable, abort_handle) = abortable(future);
+        CancelableJoinHandle {
+            inner: runtime.spawn(abortable),
+            abort_handle,
+        }
+    }
+}
+
+impl<T> Drop for CancelableJoinHandle<T> {
+    fn drop(&mut self) {
+        self.abort_handle.abort()
+    }
+}
```

Comment:
Intersting that `AbortHandle` does not already implement `Drop`. Maybe this could be one for future tokio inclusion/discussion?

Reply:
I think that would be the more reasonable default behaviour (since that's what most `Future`s do). However, it's probably too late to change it at this point... :/

Reply:
(And to be clear, the relevant point here would probably be the `JoinHandle`, not the `AbortHandle`.